Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -160,23 +160,53 @@ void HTMLDetailsElement::attributeChanged(const QualifiedName& name, const AtomS
                 root->appendChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Closed, DetailsState::Open);
                 if (auto& name = attributeWithoutSynchronization(nameAttr); document().settings().detailsNameAttributeEnabled() && !name.isEmpty()) {
-                    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
-                    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
-                        if (&detailsElement != this && detailsElement.attributeWithoutSynchronization(nameAttr) == name)
-                            otherElementsInNameGroup.append(&detailsElement);
-                    }
-                    for (RefPtr otherDetailsElement : otherElementsInNameGroup)
+                    bool shouldFireMutationEvents = document().shouldFireMutationEvents();
+                    document().setShouldFireMutationEvents(false);
+                    for (RefPtr otherDetailsElement : otherElementsInNameGroup())
                         otherDetailsElement->removeAttribute(openAttr);
+                    document().setShouldFireMutationEvents(shouldFireMutationEvents);
                 }
             } else {
                 root->removeChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Open, DetailsState::Closed);
             }
         }
-    } else
+    } else {
+        ensureDetailsExclusivityAfterMutation();
         HTMLElement::attributeChanged(name, oldValue, newValue, attributeModificationReason);
+    }
+}
+
+Node::InsertedIntoAncestorResult HTMLDetailsElement::insertedIntoAncestor(InsertionType insertionType, ContainerNode& parentOfInsertedTree)
+{
+    ensureDetailsExclusivityAfterMutation();
+    return HTMLElement::insertedIntoAncestor(insertionType, parentOfInsertedTree);
+}
+
+Vector<RefPtr<HTMLDetailsElement>> HTMLDetailsElement::otherElementsInNameGroup()
+{
+    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
+    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
+        if (&detailsElement != this && detailsElement.attributeWithoutSynchronization(nameAttr) == attributeWithoutSynchronization(nameAttr))
+            otherElementsInNameGroup.append(&detailsElement);
+    }
+    return otherElementsInNameGroup;
 }
 
+void HTMLDetailsElement::ensureDetailsExclusivityAfterMutation()
+{
+    if (auto& name = attributeWithoutSynchronization(nameAttr); m_isOpen && !name.isEmpty() && document().settings().detailsNameAttributeEnabled()) {
```

Comment:
Seems slightly wasteful to do the hash table lookup for the `name` attribute even if `m_isOpen` is false.

Reply:
> Seems slightly wasteful to do the hash table lookup for the `name` attribute even if `m_isOpen` is false.

I guess this is moot now that that `m_isOpen`  is gone? Anyway, I did otherwise switch the order such that `document().settings().detailsNameAttributeEnabled()` preference check is called before the lookups for the `name` attribute and (now `open` attribute) — because that’s how most of the existing code for similar cases seems to order the checks. (And I guess maybe in general a preference check is less expensive than an attribute lookup?)